crates/sui-move/tests/new_tests.rs

```diff
@@ -0,0 +1,385 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow;
+use cli_sandbox::Project;
+use insta::{assert_debug_snapshot, assert_snapshot};
+use insta_cmd::{assert_cmd_snapshot, get_cargo_bin};
+use std::io::Read;
+use std::path::Path;
+use std::{fs, fs::File, path::PathBuf, process::Command};
+
+/// Note: tests use [insta] and [insta-cmd]; you can review and update expected test results using
+/// cargo insta test --review
+
+/// # Test infrastructure //////////////////////////////////////////////////////////////////////////
+
+/// create a `sui-move` Command that runs in a new temporary directory
+fn sui_move() -> (Project, Command) {
+    let project = Project::new().unwrap();
+    let mut cmd = Command::new(get_cargo_bin("sui-move"));
+    cmd.current_dir(project.path());
+    (project, cmd)
+}
+
+/// create a file in `[proj]/[path]` containing `[contents]`
+fn new_file<P: AsRef<Path>>(proj: &mut Project, path: P, contents: &str) -> anyhow::Result<()> {
+    let fullpath = proj.path().join(&path);
+    let parent = fullpath
+        .parent()
+        .ok_or_else(|| anyhow::anyhow!("joined path is expected to have a parent"))?;
+    fs::create_dir_all(parent)?;
+    proj.new_file(path, contents)
+}
+
+/// return the contents of the file `[proj]/[path]`
+fn slurp_file<P: AsRef<Path>>(proj: &Project, path: P) -> anyhow::Result<String> {
+    let mut buf: String = String::new();
```

Comment:
There is a [`std::fs::read_to_string`](https://doc.rust-lang.org/beta/std/fs/fn.read_to_string.html)